src/tests/pools/D3MAaveV3NoSupplyCapTypePool.t.sol

```diff
@@ -0,0 +1,329 @@
+
+        hub = address(new FakeHub(vat));
+        end = FakeHub(hub).end();
+
+        d3mTestPool = address(new D3MAaveV3NoSupplyCapTypePool("", hub, address(dai), address(aavePool)));
+    }
+
+    function test_sets_dai_value() public {
+        assertEq(address(D3MAaveV3NoSupplyCapTypePool(d3mTestPool).dai()), address(dai));
+    }
+
+    function test_can_file_king() public {
+        assertEq(D3MAaveV3NoSupplyCapTypePool(d3mTestPool).king(), address(0));
+
+        D3MAaveV3NoSupplyCapTypePool(d3mTestPool).file("king", address(123));
+
+        assertEq(D3MAaveV3NoSupplyCapTypePool(d3mTestPool).king(), address(123));
+    }
+
+    function test_cannot_file_king_no_auth() public {
+        D3MAaveV3NoSupplyCapTypePool(d3mTestPool).deny(address(this));
+        assertRevert(d3mTestPool, abi.encodeWithSignature("file(bytes32,address)", bytes32("king"), address(123)), "D3MAaveV3NoSupplyCapTypePool/not-authorized");
+    }
+
+    function test_cannot_file_king_vat_caged() public {
+        FakeVat(vat).cage();
+        assertRevert(d3mTestPool, abi.encodeWithSignature("file(bytes32,address)", bytes32("king"), address(123)), "D3MAaveV3NoSupplyCapTypePool/no-file-during-shutdown");
+    }
+
+    function test_deposit_calls_lending_pool_deposit() public {
+        D3MTestGem(address(adai)).rely(address(aavePool));
+        vm.prank(hub);
+        D3MAaveV3NoSupplyCapTypePool(d3mTestPool).deposit(1);
+        (address asset, uint256 amt, address dst, uint256 code) = FakeLendingPool(address(aavePool)).lastDeposit();
+        assertEq(asset, address(dai));
+        assertEq(amt, 1);
+        assertEq(dst, d3mTestPool);
+        assertEq(code, 0);
+    }
+
+    function test_withdraw_calls_lending_pool_withdraw() public {
+        // make sure we have Dai to withdraw
+        D3MTestGem(address(dai)).mint(address(aavePool), 1);
+
+        vm.prank(hub);
+        D3MAaveV3NoSupplyCapTypePool(d3mTestPool).withdraw(1);
+        (address asset, uint256 amt, address dst) = FakeLendingPool(address(aavePool)).lastWithdraw();
+        assertEq(asset, address(dai));
+        assertEq(amt, 1);
+        assertEq(dst, hub);
+    }
+
+    function test_withdraw_calls_lending_pool_withdraw_vat_caged() public {
+        // make sure we have Dai to withdraw
+        D3MTestGem(address(dai)).mint(address(aavePool), 1);
+
+        FakeVat(vat).cage();
+        vm.prank(hub);
+        D3MAaveV3NoSupplyCapTypePool(d3mTestPool).withdraw(1);
+        (address asset, uint256 amt, address dst) = FakeLendingPool(address(aavePool)).lastWithdraw();
+        assertEq(asset, address(dai));
+        assertEq(amt, 1);
+        assertEq(dst, hub);
+    }
+
+    function test_collect_claims_for_king() public {
+        address king = address(123);
+        address rewardsClaimer = adai.getIncentivesController();
+        D3MAaveV3NoSupplyCapTypePool(d3mTestPool).file("king", king);
+
+        D3MAaveV3NoSupplyCapTypePool(d3mTestPool).collect(address(123));
```

Comment:
It feels a bit confusing the `reward` is the same address than the `king`, I know it doesn't matter for the test to pass, but semantically is confusing. I'd just use another random address.